### PR TITLE
Provide Simulator.initialise to perform initialisation.

### PR DESCRIPTION
**Pull request by jetuk**
Useful method if one wishes to run initialise without doing the
simulation. Optional keyword argument to start can skip
initialisation if required. User must keep track of whether this
has been done.

### CODE REVIEW

File: pynsim/simulators/simulator.py [hunk 1]
```diff
@@ -91,7 +91,21 @@ def __repr__(self):
...
+            logging.critical("No network to simulate!")
+            return
+
+        if len(self.timesteps) == 0:
+            logging.critical("No timesteps specified!")
+            return
+
+        for engine in self.engines:
+            logging.debug("Setting up engine %s", engine.name)
+            engine.initialise()
```
Comment (knoxsp) [inline pynsim/simulators/simulator.py:105]: Is it intended that by setting initalise to false, none of the engines will be initialised?

File: pynsim/simulators/simulator.py [hunk 2]
```diff
@@ -107,17 +121,8 @@ def tqdm(iterable, **kwargs):
...
-            return
-
-        if len(self.timesteps) == 0:
-            logging.critical("No timesteps specified!")
-            return
-
-        for engine in self.engines:
-            logging.debug("Setting up engine %s", engine.name)
-            engine.initialise()
+        if initialise:
```
Comment (knoxsp) [inline pynsim/simulators/simulator.py]: Can you be more explicit about te type here? for example 
```python
if initialise is True:
```

File: pynsim/simulators/simulator.py (hunk 1 above)
Comment (knoxsp) [reply pynsim/simulators/simulator.py:105]: Incidenatlly, I have merged the previous MR into master, so the logging criticals are now runtime errors.

Comment (jetuk) [reply pynsim/simulators/simulator.py:105]: The `initialise` keyword is only provided to the `start` method. It's meant to allow you to control the initialisation step separately to the running of the simulation step. If you've called `.initialise()` manually then you don't want to have that process repeated during `start()`.

**Comment by jetuk** [comment]
One thing I realised writing this is whether we need a separate `reset` process rather for a rerun rather than a reload?

File: pynsim/simulators/simulator.py (hunk 1 above)
Comment (knoxsp) [reply pynsim/simulators/simulator.py:105]: OK, understood. that's fine.

**Comment by knoxsp** [comment]
I've made a separte issue relating to the 'reset' / 'reload' question. comment on there as appropriate.

**Comment by aphurford** [comment]
Not sure I need to be subscribed to this thread, can one of you remove me?

Cheers
A

**Pull request merged**